### PR TITLE
Add new property to disable deprecations

### PR DESCRIPTION
**Pull request by IchHabRecht**

### CODE REVIEW

File: src/TestingFramework/TestCase/FunctionalTestCase.php [hunk 1]
```diff
@@ -111,6 +111,13 @@
      */
     protected $configurationToUseInTestInstance = [];
 
+    /**
+     * If true, deprecation errors are not handled.
+     *
+     * @var bool
+     */
+    protected $disableDeprecations = false;
```
Comment (oliverklee) [inline src/TestingFramework/TestCase/FunctionalTestCase.php:119]: I propose removing the negative, i.e., changing it to "enableDeprecations" (with a default value of true). Enabling a flag to disable something is confusing and error prone.

Comment (IchHabRecht) [reply src/TestingFramework/TestCase/FunctionalTestCase.php:119]: To be honest I don't get the difference between enabling a disabled option and disable an enabled option. Would you mind to explain why you think the first one is confusing and the second one isn't?

Comment (oliverklee) [reply src/TestingFramework/TestCase/FunctionalTestCase.php:119]: I'll do my best. :-)

You can see the "cofusingness" as a combination of "number of mental steps needed to understand" and "level of contradiction"/"intuitiveness"

Number of mental steps:
- not reversed: 1. enable (flag) to 2. enable (feature)
- reversed: 1. enable (flag) to 2. (not) 3. enable (feature)

Level of contradiction:
- not reversed: flag enabled = feature enabled, flag disabled = feature disabled
- reversed: flag enabled = feature disabled, flag disabled = feature enabled

This also has to do with the mind grasping a statement first (intutively), and having trouble directly integrating a negation. For example, people get stressed out far more if someone tells them "I don't want to tell you that there is a problem." vs. "I want to tell you that everything works fine." (although both statements basically say the same, but the gut first reacts to the statement, ignoring the negation).

For background, I recommend this article:
https://schneide.wordpress.com/2014/08/03/dont-ever-not-avoid-negative-logic/

Comment (oliverklee) [reply src/TestingFramework/TestCase/FunctionalTestCase.php:119]: And another article on the psychological aspects of negations in general (not programming-related): https://www.speechcare.de/news/wie-verneinungen-unser-gehirn-verwirren/

**Comment by IchHabRecht** [comment]
After some more investigation, this patch isn't needed and therefore be abandoned. The core needs to be fixed to catch deprecation errors correctly. Furthermore the testing-framework will get some compat patches for 9.0 and 9.1 support.

File: src/TestingFramework/TestCase/FunctionalTestCase.php (hunk 1 above)
Comment (IchHabRecht) [reply src/TestingFramework/TestCase/FunctionalTestCase.php:119]: Hi @oliverklee,

Thanks for sharing your point and the article!

**Pull request closed without merging**